.github/workflows/scripts/release-branch.sh

```diff
@@ -0,0 +1,163 @@
+#!/bin/bash -ex
+#
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
+# --- Configuration ---
+RELEASE_SERIES="$1" # e.g., v0.85.x
+PREPARE_RELEASE_COMMIT_HASH="$2" # Optional: Specific commit hash for "Prepare release"
+UPSTREAM_REMOTE_NAME=${UPSTREAM_REMOTE_NAME:-"upstream"} # Your upstream remote name for open-telemetry/opentelemetry-collector
+MAIN_BRANCH_NAME=${MAIN_BRANCH_NAME:-"main"}
+LOCAL_MAIN_BRANCH_NAME=${LOCAL_MAIN_BRANCH_NAME:-"${MAIN_BRANCH_NAME}"}
+# These variables are only used if git user.name and git user.email are not configured
+GIT_CONFIG_USER_NAME=${GIT_CONFIG_USER_NAME:-"opentelemetrybot"}
+GIT_CONFIG_USER_EMAIL=${GIT_CONFIG_USER_EMAIL:-"107717825+opentelemetrybot@users.noreply.github.com"}
+
+# --- Helper Functions ---
+check_command_success() {
+  if [[ $? -ne 0 ]]; then
+    echo "Error: Command failed - '$1'. See output above for details."
+    exit 1
+  fi
+}
+
+# --- Validate Input ---
+if [[ -z "$RELEASE_SERIES" ]]; then
+  echo "Error: Release series not provided."
+  echo "Usage: $0 <release-series> [prepare-release-commit-hash]"
+  echo "Example: $0 v0.85.x"
+  echo "Example: $0 v0.85.x a1b2c3d4"
+  exit 1
+fi
+
+RELEASE_BRANCH_NAME="release/${RELEASE_SERIES}"
+
+echo "Automating Release Steps for: ${RELEASE_SERIES}"
+echo "Release Branch Name: ${RELEASE_BRANCH_NAME}"
+if [[ -n "$PREPARE_RELEASE_COMMIT_HASH" ]]; then
+  echo "Will use specific 'Prepare release' commit: ${PREPARE_RELEASE_COMMIT_HASH}"
+else
+  echo "Will use HEAD of '${UPSTREAM_REMOTE_NAME}/${MAIN_BRANCH_NAME}' (after fetch/rebase) as 'Prepare release' commit."
+fi
+echo "Upstream Remote: ${UPSTREAM_REMOTE_NAME}"
+echo "--------------------------------------------------"
+
+# --- Step 4: Checkout main, Pull, Create/Update and Push Release Branch ---
+echo ""
+echo "=== Step 4: Preparing and Pushing Release Branch ==="
+
+# 1. Checkout main
+echo "1. Checking out '${MAIN_BRANCH_NAME}' branch..."
+
+# Check if Git user.name is configured
+if ! git config user.name > /dev/null 2>&1; then
+  echo "Git user.name not configured. Setting to ${GIT_CONFIG_USER_NAME}"
+  git config user.name "${GIT_CONFIG_USER_NAME}"
+else
+  echo "Git user.name already configured: $(git config user.name)"
+fi
+
+# Check if Git user.email is configured
+if ! git config user.email > /dev/null 2>&1; then
+  echo "Git user.email not configured. Setting to ${GIT_CONFIG_USER_EMAIL}"
+  git config user.email "${GIT_CONFIG_USER_EMAIL}"
+else
+  echo "Git user.email already configured: $(git config user.email)"
+fi
+
+git checkout "${LOCAL_MAIN_BRANCH_NAME}"
+check_command_success "git checkout ${LOCAL_MAIN_BRANCH_NAME}"
+
+# 2. Fetch from upstream (updates remote-tracking branches including potential existing release branch)
+echo "2. Fetching from '${UPSTREAM_REMOTE_NAME}'..."
+git fetch "${UPSTREAM_REMOTE_NAME}"
```

Comment:
You can use [`set -x`](https://gist.github.com/mohanpedala/1e2ff5661761d3abd0385e8223e16425#set--x) to avoid having to do the `echo` statements (not exactly the same, but I think seeing `git fetch` printed is good enough as a log here for example)